src/server/models/order.ts

```diff
@@ -32,30 +32,107 @@ class Order extends Model<Order> {
   @BelongsToMany(() => Product, () => OrdersProducts)
   products!: Product[]
 
-  static addToCart(userId: number, productId: number) {
-    return Order.findAll({
+  static addToCart(userId: number, productId: number, quantity: number = 1) {
+    console.log(`Add to Cart: User|Product|Qty: ${userId}|${productId}|${quantity}`)
+    // 1. no orders: make a first cart
+    // 2. there are orders, but no cart: make a cart
+    // 3. there is a cart, find one
+    return Order.findOne({
       where: {
         userId,
         status: Status.Cart
       }
     })
-      .then(async orders => {
+      .then(async orderCart => {
         let cart
-        if (orders.length === 0) {
-          console.log('cannot find order: creating one')
+        if (!orderCart) {
+          console.log(`cannot find cart for userId ${userId}: creating one`)
           cart = await Order.create({ userId, status: Status.Cart })
-        } else {
-          cart = orders.find(order => order.status === Status.Cart)
-          if (!cart) {
-            console.log('cannot find cart: creating one')
-            cart = await Order.create({ userId, status: Status.Cart })
-          }
-        }
+        } else cart = orderCart
+        // else {
+        //   cart = orders.find(order => order.status === Status.Cart)
+        //   if (!cart) {
+        //     console.log(`cannot find cart for userId : creating one`)
+        //     cart = await Order.create({ userId, status: Status.Cart })
+        //   }
+        // }
         return cart
       })
-      .then(cart => OrdersProducts.create({ orderId: cart.id, productId: productId }))
-      .then(lineItem => lineItem.orderId)
+      // 4. find existing ordersProduct instance and increase quantity by quantity.
+      // 5. if no existing ordersProduct instance, create instance.
+      .then(cart => {
+        // cart = order with status cart
+        return OrdersProducts.findAll({
+          where: {
+            orderId: cart.id,
+            status: Status.Cart
+          }
+        })
+          .then(async (userOrderCart) => {
+            // userOrderCart is the orderProducts for the user's cart.
+            // If the productId exists, add quantity, else create product with that qty.
+            let foundProductFromCart = userOrderCart.find(orderProdLineItem => orderProdLineItem.productId === productId)
+            if (!foundProductFromCart) {
+              // product does not exist in cart. add product to order
+              return await OrdersProducts.create({ orderId: cart.id, productId: productId, quantity: quantity })
+            } else {
+              // product exists in cart. increase quantity
+              return await foundProductFromCart.update({ quantity: foundProductFromCart.quantity + quantity })
+            }
+          })
+          .then(() => cart.id)
```

Comment:
Have you tested this? Can it really find cart? I thought it would be out of scope by now

Reply:
Yes, this is tested. Console logged where cart.id was initialized and then at the end where it is returned results in the same value.